Graph Panel: use FieldConfig for displayName and DataLinks

ryantxu opened the pull request:
The field config options for setting names is super important for DataSources that do not (and should not!) implement their own field name alias structure -- for example:  the new Azure Monitor support and flux in InfluxDB

This PR takes a small step towards using FieldConfig to define the series settings.

![image](https://user-images.githubusercontent.com/705951/86066184-5a04a980-ba26-11ea-8681-b4043ed194f3.png)

TODO:

- [x] Is this OK to do as a small step?  Is it too confusing to have "Field override" and "Series overrieds"
- [x] Better migration tests -- make sure FieldConfigs migrate OK

The other most obvious setting that we should *really* try to make work is units/decimals -- the details get complicated because existing settings/config is actually about the "axis"

ryantxu committed:
graph with new controls

ryantxu committed:
graph with new controls

dprokop committed:
Fix categories

dprokop committed:
Merge branch 'master' into graph-field-config

ryantxu committed:
use config for DataLinks

ryantxu commented on public/app/features/dashboard/state/PanelModel.ts:
```diff
@@ -334,7 +334,7 @@ export class PanelModel implements DataConfigSource {
     this.plugin = plugin;
 
-    if (plugin.panel && plugin.onPanelMigration) {
+    if (plugin.onPanelMigration) {
```
The real change here is to allow migrations on angular panels *and* react panels

ryantxu committed:
Merge remote-tracking branch 'origin/master' into graph-field-config

ryantxu committed:
fix test imports

ryantxu commented:
@torkelo can you take a quick look at this?  Do you think we also need to figure out how to use `units`?  or is this enough for now?

Units is complicated because they are currently linked to the axis settings

RichiH committed:
Chore: Fix spelling (#25905)

dprokop committed:
grafana-ui: Make output chunks correct (#25922)

ivanahuckova committed:
Explore/Loki: Show results of instant queries only in table and time series only in graph (#25845)

* Show results of instant queries only in table and rest in graph

* Add type to QueryResultMeta

* Update log row hover background only if context is not open

* Revert "Update log row hover background only if context is not open"

This reverts commit 144197c9540565273515a6d7dd38d3e12a436f05.

ivanahuckova committed:
Explore/Loki: Fix field type in table for instant queries (#25907)

* Update field type for instant queries to number

* Remove fieldtype as in prometheus

* Revert "Remove fieldtype as in prometheus"

This reverts commit f887ad1125deb1053e33f7ef385d5f518ac27892.

aknuds1 committed:
Chore: Disable scopelint for tests (#25923)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

dprokop committed:
grafana/ui: Fix storybook dev (#25928)

* Minimize css and js only for storybook build

* Update CodeEditor story, make it internal

* Update storybook webpack config

mckn committed:
TimeZonePicker: added possibility to toggle if internal time zones should be included or not. (#25934)

* made some small adjustments after feedback.

* made the flag optional.

aknuds1 committed:
Docker: Upgrade to base on Alpine 3.12 (#25936)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aocenas committed:
DataLinks: Add internal links in table and allow custom query (#25613)

* Add internal links in table and with custom query

* Add specific types for internal and external link

* Change the datalink types to be more backward compatible

* Refactor the link utils for explore

* Add internal linking to table panels

* Fix derived field condition

* Prettify

* Add and fix tests

* Prettify

* Fix imports and tests

* Remove unused type

* Update packages/grafana-data/src/types/explore.ts

Co-authored-by: Dominik Prokop <dominik.prokop@grafana.com>

* Update packages/grafana-data/src/types/explore.ts

Co-authored-by: Dominik Prokop <dominik.prokop@grafana.com>

Co-authored-by: Dominik Prokop <dominik.prokop@grafana.com>

aknuds1 committed:
CircleCI: Upgrade grabpl (#25939)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

nabokihms committed:
Provisioning: Use folders structure from the file system to create desired folders in dashboard provisioning. (#23117)

Fixes #12016

stevenvachon committed:
Added options to customize chart data listening (#25918)

ivanahuckova committed:
Explore: Remove legend formatting when switching from panel to Explore (#25848)

* Remove legend formatting when switching from panel to explore

* Add test coverage

ivanahuckova committed:
Query history: Add search for query history and starred queries (#25747)

* Add searching functionality to Queries tab

* Implement search for starred queries

* Update tests and add test coverage for searching

* Changes based on provided feedback

* Remove redundant storing of results

* Update naming of onSearchFilterChange function in tests

* Update filterQueriesBySearchFilter function

* Update filterQueriesBySearchFilter function

* Debounce search

* Remove console.log()

* Use useDebouncedCallback

* Update tests, remove redundant props

* When in starred, search only in starred

* Add timefilter as dependency

* Remove redundant dependency, refactor

Clarity-89 committed:
Grafana-UI: Add FileUpload (#25835)

* Grafana UI: Setup component

* Grafana UI: DashboardFileUpload => FileUpload

* Grafana UI: Expand docs

* Grafana UI: Add className

* Grafana UI: Update import

* Grafana UI: Clarify props

* Update packages/grafana-ui/src/components/FileUpload/FileUpload.tsx

Co-authored-by: Tobias Skarhed <1438972+tskarhed@users.noreply.github.com>

* Grafana UI: Style icon

* Grafana UI: Show uploaded file name

* Grafana UI: Add tests

* Grafana UI: Add useStyles + useCallback

* Grafana UI: Remove stylesFactory

* Grafana UI: Revert to useTheme

Co-authored-by: Tobias Skarhed <1438972+tskarhed@users.noreply.github.com>

peterholmberg committed:
Fix: Value mappings match against string values (#25929)

* Use standard input for value mappings value

* add test and case for matching string values

* re add number on rangemap

* remove FieldTypeNumber required

torkelo committed:
Docs: Changelog update 7.05 & latest.json update (#25941)

* Docs: Changelog update for 7.0.5

* Update latest.json

* Fixed alignment

papagian committed:
Stackdriver: Rename Stackdriver to Google Cloud Monitoring (#25807)

* Update backend

* Update frontend

* Keep old plugin id

* Update docs

* Place doc images to a new directory

* Legacy support for stackdriver-auto alignment

* Consistent plugin name

* Apply suggestions from code review

Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>
Co-authored-by: Tobias Skarhed <1438972+tskarhed@users.noreply.github.com>

* Update docs

* Update public/app/plugins/datasource/cloud-monitoring/README.md

Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>

* Add reference to the data source formerly being named Stackdriver

* Update pkg/models/datasource.go

Co-authored-by: Carl Bergquist <carl@grafana.com>

* Fix gofmt

Co-authored-by: Marcus Efraimsson <marcus.efraimsson@gmail.com>
Co-authored-by: Tobias Skarhed <1438972+tskarhed@users.noreply.github.com>
Co-authored-by: Carl Bergquist <carl@grafana.com>

torkelo reviewed (commented):
I took this for a spin and it works :) 

Sure a bit odd to only have these two settings available via Field & Overrides tab, that is the main concern. 

But I think it is fine.

ivanahuckova committed:
Fix strict null errors that resulted from search (#25950)

ryantxu committed:
Arrow: when exporting arrow use the calculated names (#25871)

torkelo committed:
PanelHeader: Fixed loading icon position (#25949)

* PanelHeader: Fixed loading icon position

* revert back to text weak color

kylebrandt committed:
Azure/Log Analytics: return more error information (#25937)

When there is an error message in the HTTP response, this copies that error so it is relayed to the user by being shown in the frontend.

This also makes it so when there is an error the interpolated query string (ExecutedQueryString) is returned so users can see it in the query inspector.

kylebrandt committed:
Azure: Multiple dimension support for Azure Monitor Service (#25947)

Azure Monitor (metrics) support multiple dimensions instead of just one.

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

oddlittlebird committed:
Docs: Update alert-list-panel.md (#25779)

Fixed minor typo and format updates

oddlittlebird committed:
Docs: Update add-a-panel.md (#25780)

bergquist committed:
Linting fixes for the provisioning package (#25690)

stevenvachon committed:
Chore: whatwg-fetch version bump (#25952)

* Use latest fetch polyfill/shim

* Updated lockfile

ryantxu committed:
Monaco: option to hide line numbers (#25920)

ryantxu committed:
Merge remote-tracking branch 'origin/master' into graph-field-config

oddlittlebird committed:
Docs: Reorganize reference topics (#25915)

* moved annotations.md

* Update annotations.md

* moved dashboard_folders.md

* moved dashboard_history.md

* moved playlist.md

* moved search.md

jsoref committed:
docs: spelling (#24383)

* Docs: spelling

* Docs: spelling (contribute/ directory)

* Drop whats-new-in changes

ryantxu committed:
migrations test

ryantxu committed:
Merge remote-tracking branch 'origin/master' into graph-field-config

ryantxu committed:
migrations test

ryantxu commented:
Sorry for the git history...  but the files should be OK now

ryantxu commented:
@torkelo -- can this get merged in beta1?
